packages/plugins/ens/src/index.js

```diff
@@ -132,32 +132,40 @@ class ENS {
   }
 
   async addArtifactFile(_params, cb) {
-    this.getEnsConfig((err, config) => {
-      this.events.request("pipeline:register", {
-        path: [this.config.embarkConfig.generationDir, 'config'],
-        file: 'namesystem.json',
-        format: 'json',
-        content: Object.assign({}, this.embark.config.namesystemConfig, config)
-      }, cb);
-    });
+    if (this.embark.config.blockchainConfig.enabled) {
+      this.getEnsConfig((err, config) => {
+        this.events.request("pipeline:register", {
+          path: [this.config.embarkConfig.generationDir, 'config'],
+          file: 'namesystem.json',
+          format: 'json',
+          content: Object.assign({}, this.embark.config.namesystemConfig, config)
+        }, cb);
+      });
+    } else {
+      cb();
+    }
   }
 
   async setProviderAndRegisterDomains(cb = (() => {})) {
-    this.getEnsConfig(async (err, config) => {
-      if (this.doSetENSProvider) {
-        this.setupEmbarkJS(config);
-      }
+    if (this.embark.config.blockchainConfig.enabled) {
+      this.getEnsConfig(async (err, config) => {
+        if (this.doSetENSProvider) {
+          this.setupEmbarkJS(config);
+        }
 
-      const web3 = await this.web3;
-      const networkId = await web3.eth.net.getId();
-      const isKnownNetwork = Boolean(ensContractAddresses[networkId]);
-      const shouldRegisterSubdomain = this.config.namesystemConfig.register && this.config.namesystemConfig.register.subdomains && Object.keys(this.config.namesystemConfig.register.subdomains).length;
-      if (isKnownNetwork || !shouldRegisterSubdomain) {
-        return cb();
-      }
+        const web3 = await this.web3;
+        const networkId = await web3.eth.net.getId();
```

Comment:
Since we have `await`s it's probably worth putting in `try/catch` and callback the error in case. Otherwise we could crash